src/libs/Navigation/AppNavigator/DrawerNavigator/index.js

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import withWindowDimensions from '../../../../components/withWindowDimensions';
+import BaseDrawerNavigator from './BaseDrawerNavigator';
+import propTypes from './drawerNavigatorPropTypes';
+
+const DrawerNavigator = props => (
+    <BaseDrawerNavigator
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...props}
+
+        // On mobile web, the non legacy version of the DrawerNavigator breaks the layout on Desktop while resizing
```

Comment:
Hmm this comment doesn't make much sense to me, and I reading this wrong?

> On mobile web, the non legacy version of the DrawerNavigator breaks the layout on Desktop...

We're only using the legacy version when the screen is small (`props.isSmallScreenWidth` is true) which is only applicable:
1. On desktop & web when the window is sized small
2. On mobile web always

Right? So the dynamic legacy implementation is needed only for web (non-mobile) & desktop right?

Reply:
There are two issues here. based on the drawer version.
Legacy implementation:
1. WEB (> 800): breaks the layout on resize 
2. Mobile Web: Works well.

Non-Legacy(modern) implementation
1. WEB (> 800): Works well.
2. Mobile Web: Drawer does not work, Closing and opening of a drawer do not work.


So use Legacy for M-web and Modern on Desktop.
On native use Modern for all cases.

Reply:
Just to be clear, when you mention WEB or Desktop by themselves, do you mean web (non-mobile web) AND desktop?

Ex:
- `**WEB** (> 800): breaks the layout on resize` - does this apply to Desktop too?
- `So use Legacy for M-web and Modern on **Desktop**.` - does "Desktop" apply to web (non mobile web) too?

Reply:
WEB > 800 also applies to Desktop and Desktop also applies to WEB.

Reply:
What do you think about this: https://github.com/Expensify/App/pull/5745/files#r739667981? Is it accurate?

Reply:
Let's get this merged quickly. Any thoughts @Beamanator ?